File: src/main/kotlin/dev/jefy/connectpro/marketplace/applicaion/dtos/JobApplicationResponseForUser.java
package dev.jefy.connectpro.marketplace.applicaion.dtos;


import java.time.Instant;

import dev.jefy.connectpro.marketplace.domain.model.JobApplication;
import dev.jefy.connectpro.shared.domain.vo.JobApplicationStatus;

/**
 * @author Jôph Yamba
 */
public record JobApplicationResponseForUser(
        JobPostListingResponse jobPost,
        String motivation,
        Instant appliedAt,
        JobApplicationStatus status
) {
    public static JobApplicationResponseForUser fromDomain(
            JobApplication application,
            JobPostListingResponse jobPostListingResponse
    ) {
        return new JobApplicationResponseForUser(
                jobPostListingResponse,
                application.getMotivation(),
                application.getAppliedAt(),
                application.getStatus()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/marketplace/application/dtos/JobApplicationResponseForUser.kt
package dev.jefy.connectpro.marketplace.application.dtos

import dev.jefy.connectpro.marketplace.application.dtos.JobPostListingResponse
import dev.jefy.connectpro.marketplace.domain.model.JobApplication
import dev.jefy.connectpro.shared.domain.vo.JobApplicationStatus
import java.time.Instant


/**
 * @author Jôph Yamba
 */

data class JobApplicationResponseForUser(
    val jobPost: JobPostListingResponse,
    val motivation: String,
    val appliedAt: Instant,
    val status: JobApplicationStatus
)
fun JobApplication.toUserResponse(jobPostListing: JobPostListingResponse) = JobApplicationResponseForUser(
    jobPost = jobPostListing,
    motivation = this.motivation,
    appliedAt = this.appliedAt,
    status = this.status
)
